sysl2/sysl/diagutil.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"bytes"
+	"compress/zlib"
+	"fmt"
+	"image/jpeg"
+	"image/png"
+	"net/http"
+	"os"
+
+	"github.com/sirupsen/logrus"
+)
+
+func OutputPlantuml(mode, url string, umlInput []byte) {
+	if url == "" {
+		url = os.Getenv("SYSL_PLANTUML")
+	}
+	url = fmt.Sprintf("%s/%s/%s", url, mode, DeflateAndEncode(umlInput))
+	resp, err := http.Get(url)
+	if err != nil {
+		logrus.Errorf("Unable to create http request to %s, Error:", url, err.Error())
+	}
+	defer resp.Body.Close()
+
+	img, err := png.Decode(resp.Body)
+	if err != nil {
+		logrus.Errorf("Unable to create image from response, Error:", err.Error())
+	}
+
+	f, err := os.Create("_.jpg")
+	if err != nil {
+		logrus.Errorf("Unable to create file, Error:", err.Error())
+	}
+	defer f.Close()
+	jpeg.Encode(f, img, nil)
+}
+
+func DeflateAndEncode(text []byte) string {
+	var buf bytes.Buffer
+	zw, err := zlib.NewWriterLevel(&buf, zlib.BestCompression)
+	if err != nil {
+		logrus.Errorf("Unable to encode []byte, Error:", err.Error())
+	}
+	zw.Write(text)
+	zw.Flush()
+	zw.Close()
+	return encode(buf.Bytes())
+}
+
+func encode(data []byte) string {
+	var buf bytes.Buffer
+	for i := 0; i < len(data); i += 3 {
+		if i+2 == len(data) {
+			encode3bytes(&buf, data[i], data[i+1], 0)
+		} else if i+1 == len(data) {
+			encode3bytes(&buf, data[i], 0, 0)
+		} else {
+			encode3bytes(&buf, data[i], data[i+1], data[i+2])
+		}
+	}
+	return buf.String()
+}
+
+func encode3bytes(buf *bytes.Buffer, b1, b2, b3 byte) {
```

Comment:
add comment to explain, like
3 bytes takes 24 bits. This splits 24 bits into 4 bytes of which lower 6-bit takes account.

Reply:
Fixed